pkg/image/admission/admission.go

```diff
@@ -0,0 +1,319 @@
+		ism, ok := obj.(*imageapi.ImageStreamMapping)
+		if !ok {
+			glog.V(4).Infof("%s: unable to convert obj to ImageStreamMapping: %#v", pluginName, obj)
+			return nil
+		}
+		images = map[string]imageapi.Image{ism.Image.Name: ism.Image}
+	case *imageapi.ImageStreamTag:
+		tag, ok := obj.(*imageapi.ImageStreamTag)
+		if !ok {
+			glog.V(4).Infof("%s: unable to convert obj to ImageStreamTag: %#v", pluginName, obj)
+			return nil
+		}
+
+		if tag.Tag == nil || tag.Tag.From == nil {
+			glog.V(4).Infof("%s: ignoring ist with no tag reference to follow: %v", pluginName, tag)
+			return nil
+		}
+
+		computer := imagequota.NewGenericImageStreamUsageComputer(a.osClient, true, false)
+		ref, err := computer.GetImageReferenceForObjectReference(tag.Namespace, tag.Tag.From)
+		if err != nil {
+			return err
+		}
+		image, err := computer.GetImage(ref.ID)
+		if err != nil {
+			return err
+		}
+		if image == nil {
+			return fmt.Errorf("unable to find image with ID %s", ref.ID)
+		}
+		images = map[string]imageapi.Image{image.Name: *image}
+	case *imageapi.ImageStream:
+		is, ok := obj.(*imageapi.ImageStream)
+		if !ok {
+			glog.V(4).Infof("%s: unable to convert obj to ImageStream: %#v", pluginName, obj)
+			return nil
+		}
+
+		previousIS, _, err := a.imageStreamStore.Get(is)
+		if err != nil {
+			// don't return, process all images in the IS
+			glog.V(4).Infof("%s: error retrieving previous image stream from cache: %v", pluginName, err)
+		}
+
+		if previousIS == nil {
+			images, err = a.getImageStreamImages(is, nil)
+		} else {
+			images, err = a.getImageStreamImages(is, previousIS.(*imageapi.ImageStream))
+		}
+		if err != nil {
+			return err
+		}
+	default:
+		glog.V(5).Infof("%s: recieved object that was not an ImageStreamMapping, ImageStreamTag, or ImageStream.  Ignoring", pluginName)
+		return nil
+	}
+
+	if len(images) == 0 {
+		glog.V(4).Infof("%s: unable to validate image limit ranges for %#v, no images were found", pluginName, obj)
+		return nil
+	}
+
+	errs := []error{}
+
+	for _, img := range images {
+		if value, exists := img.Annotations[imageapi.ManagedByOpenShiftAnnotation]; !exists || value != "true" {
+			glog.V(5).Infof("%s: image is not managed by openshift, ignoring", pluginName)
+			continue
+		}
+
+		// ensure metadata is loaded from the manifest, this may not be automatically filled in if
+		// an older registry is in use
+		err := imageapi.ImageWithMetadata(&img)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		for _, limit := range limitRange.Spec.Limits {
+			err := AdmitImage(img.DockerImageMetadata.Size, limit)
+			if err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
+
+	if len(errs) != 0 {
+		return utilerrors.NewAggregate(errs)
+	}
+
+	return nil
+}
+
+// getImageStreamImages gives back a slice of images that need checked against the limit range
+// based on the image stream.  It filters out preexisting images that do not need to be checked
+// by using the previousIS.
```

Comment:
The advantage of this is I get to reuse the code @miminar wrote that actually fetches the images for me as it processes the streams.  I'm not opposed to changing it if you have a better implementation but I'd rather not rewrite the loop, process, fetch portions if I can reuse them.